openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -81,6 +81,17 @@ def _is_block_shiftable(xblock):
             PersonalizedLearnerScheduleCallToAction._log_past_due_warning(type(xblock).__name__)
             is_past_due = xblock.is_past_due
 
+        # This is used to replicate the logic used in common/lib/xmodule/xmodule/vertical_block.py
+        # We only want to show the banner if there are any problems that are graded and scored.
+        # Note: we will still show the CTA at the xblock level (next to the submit button) regardless
+        # of graded and scored status
+        if category == cls.VERTICAL_BANNER:
+            graded = getattr(xblock, 'graded', False)
+            has_score = getattr(xblock, 'has_score', False)
+            weight = getattr(xblock, 'weight', 1)
+            scored = has_score and (weight is None or weight > 0)
+            return xblock.self_paced and can_attempt and is_past_due and graded and scored
```

Comment:
I opted for the two return statements over defining default values of True for graded and scored. If you have different thoughts, let me know :)

Reply:
Maybe it's worth defining a variable at the top for `xblock.self_paced and can_attempt and is_past_due` and using that in both returns?

Reply:
You know, I actually thought about that too. Was afraid of trying to come up with a name for that.

Reply:
Maybe like so?

```
can_shift = xblock.self_paced and can_attempt and is_past_due

if xxx:
    can_shift = can_shift and graded and scored

return can_shift
```